### PR TITLE
Issue #965 Extract MediaGalleryIntegration module from the MediaGalleryUi

### PR DESCRIPTION
**Pull request by karyna-t**
### Description (*)
This PR adds a new MediaGalleryIntegration module with extracted from MediaGallery files as requested.
It also adds new Unit tests for moved files, + some clean-up in docBlocks.

### Fixed Issues (if relevant)
magento/adobe-stock-integration#965: Extract MediaGalleryIntegration module from the MediaGalleryUi

### CODE REVIEW

**Comment by diazwatson** [comment]
@karyna-tsymbal-atwix thanks very much for your contribution.
To help us process this PR, could you please solve the git conflicts?
Once you finish please ping in the comments and I will finish reviewing it

**Comment by karyna-t** [comment]
Hi @diazwatson , I resolved git conflicts. However there are some Static tests issues related to my PR, and I'm not sure what's the problem. Can you please advise how to fix it?

**Comment by diazwatson** [comment]
Hi @karyna-tsymbal-atwix this is what I see

## Static Tests
### Magento\Test\Integrity\DependencyTest
```
Passed: 16958, Failed: 1, Incomplete: 0, Skipped: 0.
Data set: app/code/Magento/MediaGalleryIntegration/Plugin/UpdateOpenDialogUrl.php
Module Magento\MediaGalleryIntegration has undeclared dependencies: hard [Magento\MediaGalleryUi]
/var/www/html/dev/tests/static/testsuite/Magento/Test/Integrity/DependencyTest.php:373
/var/www/html/lib/internal/Magento/Framework/App/Utility/AggregateInvoker.php:56
/var/www/html/dev/tests/static/testsuite/Magento/Test/Integrity/DependencyTest.php:376
```

### Magento\Test\Integrity\ComposerTest
 ```
Passed: 343, Failed: 2, Incomplete: 0, Skipped: 0.
Data set: /var/www/html/app/code/Magento/MediaGalleryIntegration
The following dependencies are missing in root 'composer.json', while declared in child components.
Consider adding them to 'require-dev' section (if needed for child components only), to 'replace' section (if they are present in the project), to 'require' section (if needed for the skeleton).
'magento/module-media-gallery-integration' depends on 'magento/module-media-gallery-ui-api'
/var/www/html/dev/tests/static/testsuite/Magento/Test/Integrity/ComposerTest.php:390
/var/www/html/dev/tests/static/testsuite/Magento/Test/Integrity/ComposerTest.php:359
/var/www/html/dev/tests/static/testsuite/Magento/Test/Integrity/ComposerTest.php:186
/var/www/html/dev/tests/static/testsuite/Magento/Test/Integrity/ComposerTest.php:97
/var/www/html/lib/internal/Magento/Framework/App/Utility/AggregateInvoker.php:56
/var/www/html/dev/tests/static/testsuite/Magento/Test/Integrity/ComposerTest.php:99
```

## Magento Health Index
```
composer.json is not equal to etalon-composer.json (dependencyComposer)
Magento/MediaGalleryIntegration/composer.json
```

File: MediaGalleryIntegration/modman [hunk 1]
```diff
@@ -0,0 +1 @@
+./ app/code/Magento/MediaGalleryIntegration
```
Comment (diazwatson) [inline MediaGalleryIntegration/modman:1]: Don't think we need to include this file

**Comment by diazwatson** [comment]
Please take a look at my other suggestions.

Thanks

**Review by diazwatson — CHANGES_REQUESTED** [review]

File: MediaGalleryIntegration/modman (hunk 1 above)
Comment (sivaschenko) [reply MediaGalleryIntegration/modman:1]: we need to do this to be able to install this module using modman

**Review by sivaschenko — APPROVED** [review]

**Pull request merged**